### PR TITLE
MAINT: signal/dlsim: avoid using interp1d; use make_interp_spline instead

### PR DESCRIPTION
**Pull request by ev-br**
Replace an `interp1d` call by a `make_interp_spline` call. Saves several transposes, too: the interpolation axis is -1 for interp1d and 0 for BSpline, and it's the latter here. 

This is not really a necessary change, I just happened to look at this code.

### CODE REVIEW

File: scipy/signal/_ltisys.py
Comment (ev-br) [reply scipy/signal/_ltisys.py:3555]: Unless I'm missing something simple, it does not extrapolate really. This change is in the `if t is not None` branch, and just a few lines above, `tout` is defined like this

```
    if t is None:
        out_samples = len(u)
        stoptime = (out_samples - 1) * system.dt
    else:
        stoptime = t[-1]
        out_samples = int(np.floor(stoptime / system.dt)) + 1
    # Pre-build output arrays
    <snip>
    tout = np.linspace(0.0, stoptime, num=out_samples)
```

so that `tout[-1] == t[-1]` and there is no extrapolation.

**Comment by larsoner** [comment]
I have not been following the CI statuses, can we ignore the two Linux Meson failures, the Wheel Builder failure, and the arm64 test failure (yikes)?

**Review by larsoner — APPROVED** [review]

**Review by tylerjereddy — APPROVED** [review]
Looks like we have a core dev approval contingent on the CI failures being unrelated.

I can confirm that the CI failures are observed elsewhere and not related. The diff also seems pretty reasonable to non-expert.

**Pull request merged**
